app/upgrades/v16/upgrades.go

```diff
@@ -69,3 +70,19 @@ func CreateUpgradeHandlerRC2(
 		return mm.RunMigrations(ctx, configurator, vm)
 	}
 }
+
+// CreateUpgradeHandlerRC3 creates an SDK upgrade handler for v16.0.0-rc3
+func CreateUpgradeHandlerRC3(
+	mm *module.Manager,
+	configurator module.Configurator,
+	ak authkeeper.AccountKeeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		logger := ctx.Logger().With("upgrade", UpgradeName)
+
+		if err := MigrateFeeCollector(ak, ctx); err != nil {
+			logger.Error("failed to migrate the fee collector", "error", err.Error())
```

Comment:
If the migration of the FeeCollector fails we will have a fork with the same error and we have to hard fork again. Isn't it better to panic? In this case, if the migration fails we just have to change the binary but the fork remains the same.